src/controllers/customercontroller.js

```diff
@@ -6,29 +6,84 @@ class CustomerController extends OrderController
             "src/views/template/customer.html",
             "#customers",
             {}
-        );
+        ).done(() => this.registerEvent());
+    }
+
+    registerEvent()
+    {
+        // TODO make the click event so it does not need to be attached to the document
+        $(document).on('click', '.customer-complete', function(e) {
+            e.preventDefault();
+
+            var id = $(this).closest("div[data-customer]").data('customer');
+
+            var customer = GAME.model.customers.filter(function (customer) {
+                return customer.id == id;
+            })[0];
+
+            let customerController = new CustomerController();
+            customerController.completeOrder(customer);
+        });
     }
 
     generateOrder()
     {
+        if (super.normalDistribution() < 1.725) {
+            return;
+        }
+
         var protoOrder = GAME.model.products.map(
             prod => {
                 return {
                     name: prod.name,
-                    value: super.randomDemandGenerator()
+                    value: super.randomDemandGenerator(
+                        prod.demand.mean,
+                        prod.demand.variance
+                    )
                 }
             }
         );
 
         var products = OrderController._makeOrder(protoOrder);
         var customer = new Customer(products);
 
+        customer.id = GAME.model.customers.length;
         GAME.model.customers.push(customer);
+
         this._updateOrderView(customer);
 
         toastr.info(Controller.l("New customer is waiting!"));
     }
 
+    completeOrder(customer)
+    {
+        var warehousecontroller = new WarehouseController();
+
+        customer.order.products.forEach(function (product) {
```

Comment:
Or a map. The thinking is that you do not so much want to 'loop over an array', you want to edit the products (as they can be fulfilled from the containers in the warehouse).

Then based on whether all products are quantity zero in the order, you may want to remove the order, so a filter seems appropriate.